lib/v3/parser.js

```diff
@@ -615,6 +655,12 @@ class Parser extends EventEmitter {
                 lastAttributeLocations = {};
             },
             onopentag: (tagName, attrs) => {
+                const isTopLevelElement = !['style', 'script'].includes(tagName) && parser._stack.length === 1;
+
+                if (isTopLevelElement && lastComment) {
```

Comment:
Just test different cases and looks like that `parser._stack.length` is always equals to `1`. The following case works fine:
```
<div>
    <div>
        <!--
        /**
        * The component description
        * @component ComponentName
        * @example
        * <ComponentName identifier={identifier} />
        */
        -->
    </div>
</div>
<div>
  <h1>Header {identifier}</h1>
</div>
```
Not sure that this should be fixed, just keep attention on this.

Reply:
Seems to me that `parser._stack.length` is 1 for root-level html elements, so it should only trigger by then, but nothing prevents us from having several top level elements.. should I introduce `level` here to keep track of first matched element excluding `style` and `script`?

Reply:
added the test case you suggested and also a level-check to prevent it from triggering more than once.

Reply:
Hm, I think that is not needed. We can just save information about first occurence of this comment. If `emitGlobalComments` was already called with not empty data we should track it and ignore all next method executions. I think that is a nice solution

Reply:
Not sure I understand 💯 % could you elaborate?
Did you see my suggested solution?

Reply:
Ok, I see your changes and now I'm understand it completelly!